src/main/java/com/rybka/Application.java

```diff
@@ -24,16 +31,39 @@ public static void main(String[] args) {
         var exchangeRateConnectorConnector = new ExchangeRateConnector(new CVurl());
 
         var exchangeSourceMap = Map.of(
-                PropertyInfo.PROPERTY_EXCHANGE_KEY, exchangeRateConnectorConnector,
-                PropertyInfo.PROPERTY_PRIME_EXCHANGE_KEY, primeExchangeRateConnector);
+                ExchangeSource.EXCHANGE.getSource(), exchangeRateConnectorConnector,
+                ExchangeSource.PRIME_EXCHANGE.getSource(), primeExchangeRateConnector);
 
         var reader = new PropertyReader(PropertyInfo.PROPERTY_FILE_NAME).getProperties();
-        var connector = exchangeSourceMap.get(reader.getProperty(PropertyInfo.PROPERTY_EXCHANGE_SOURCE));
+        var connector = exchangeSourceMap.entrySet().stream()
```

Comment:
Why do you use stream on map o_O ? 
map.get(value) will be enough and far more efficient than stream iteration. 

to throw and exception try to use Optional from java 8

Reply:
fixed